[Issue]: MIOpen failures during PyTorch smoke tests

While running [TheRock's PyTorch smoke tests](https://github.com/ROCm/TheRock/blob/main/external-builds/pytorch/smoke-tests/pytorch_smoke_test.py) on MI355, we are running into MIOpen errors during run

```
external-builds/pytorch/smoke-tests/pytorch_smoke_test.py::TestConvolutions::test_conv_transpose2d FAILED [ 90%]
external-builds/pytorch/smoke-tests/pytorch_smoke_test.py::TestConvolutions::test_conv_cudnn_nhwc_support FAILED [100%]
```

Error message is (same error for `test_conv_cudnn_nhwc_support`) :
```
____________________ TestConvolutions.test_conv_transpose2d ____________________

self = <pytorch_smoke_test.TestConvolutions object at 0x7f1c8b7e8350>

    def test_conv_transpose2d(self):
        inputs = torch.randn(1, 4, 5, 5, device="cuda")
        weights = torch.randn(4, 8, 3, 3, device="cuda")
        # Simply running any conv op exercises MIOpen and library loading.
        # On Windows, this may fail if `amd_comgr_3.dll` (from build output) is
        # used instead of `amd_comgr0605.dll` that is expected at runtime.
>       result = torch.nn.functional.conv_transpose2d(inputs, weights, padding=1)
E       RuntimeError: miopenStatusUnknownError

external-builds/pytorch/smoke-tests/pytorch_smoke_test.py:103: RuntimeError
----------------------------- Captured stderr call -----------------------------
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' naive_conv.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: naive_conv.cpp
MIOpen(HIP): Warning [BuildHip] hip runtime failed to load.
Error: Please provide architecture for which code is to be generated.
MIOpen Error: /therock/src/ml-libs/MIOpen/src/hipoc/hipoc_program.cpp:299: Code object build failed. Source: naive_conv.cpp
```

Logs are in [Test PyTorch Wheels](https://github.com/ROCm/TheRock/actions/runs/16686369057/job/47236522635)

<br/>

Steps to reproduce error (on MI355 ubuntu24 machine):

```
$ docker run -i --ipc host --group-add video --device /dev/kfd --device /dev/dri -t ghcr.io/rocm/pytorch_ubuntu_24_04_test_image:users-geomin12-pytorch-355-fix /bin/bash
$ python3 -m venv venv
$ source venv/bin/activate
$ python -m pip install --index-url=https://d25kgig7rdsyks.cloudfront.net/v2/gfx950-dcgpu torch==2.7.1+rocm7.0.0.dev0.f887e7328413f6a8fd337abf952db856c842c0dd
$ git clone https://github.com/ROCm/TheRock.git
$ cd TheRock
$ pip install -r external-builds/pytorch/requirements-test.txt
$ ./external-builds/pytorch/pytorch_torch_repo.py checkout --gitrepo-origin https://github.com/ROCm/pytorch.git --repo-hashtag release/2.7 --no-hipify --no-patch
$ pytest -v external-builds/pytorch/smoke-tests
```

--- a/patches/amd-mainline/clr/0005-Explicitly-load-libamdhip64.so.7.patch
+++ b/patches/amd-mainline/clr/0005-Explicitly-load-libamdhip64.so.7.patch
@@ -1,0 +1,39 @@
+From 71a8f02f13d25632cc579d963c79d406d55f5066 Mon Sep 17 00:00:00 2001
+From: Marius Brehler <marius.brehler@amd.com>
+Date: Wed, 6 Aug 2025 20:53:01 +0000
+Subject: [PATCH 5/5] Explicitly load `libamdhip64.so.7`
+
+---
+ hipamd/src/hip_comgr_helper.cpp | 2 +-
+ hipamd/src/hip_platform.cpp     | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/hipamd/src/hip_comgr_helper.cpp b/hipamd/src/hip_comgr_helper.cpp
+index a8ee6ea9b..4b4908a93 100644
+--- a/hipamd/src/hip_comgr_helper.cpp
++++ b/hipamd/src/hip_comgr_helper.cpp
+@@ -1151,7 +1151,7 @@ bool RTCProgram::findIsa() {
+   std::string dll_name = std::string("amdhip64_" + std::to_string(HIP_VERSION_MAJOR) + ".dll");
+   libName = dll_name.c_str();
+ #else
+-  libName = "libamdhip64.so";
++  libName = "libamdhip64.so.7";
+ #endif
+ 
+   void* handle = amd::Os::loadLibrary(libName);
+diff --git a/hipamd/src/hip_platform.cpp b/hipamd/src/hip_platform.cpp
+index da290fd5e..e01667d69 100644
+--- a/hipamd/src/hip_platform.cpp
++++ b/hipamd/src/hip_platform.cpp
+@@ -1014,7 +1014,7 @@ void* PlatformState::getDynamicLibraryHandle() {
+ #ifdef _WIN32
+   const char* libName = "amdhip64.dll";
+ #else
+-  const char* libName = "libamdhip64.so";
++  const char* libName = "libamdhip64.so.7";
+ #endif
+ 
+   dynamicLibraryHandle_ = amd::Os::loadLibrary(libName);
+-- 
+2.43.0
+

